src/tools/spin/spinquic.cpp

```diff
@@ -818,9 +814,30 @@ main(int argc, char **argv)
             auto PeerStreamCount = GetRandom((uint16_t)10);
             EXIT_ON_FAILURE(MsQuic->SetParam(Session, QUIC_PARAM_LEVEL_SESSION, QUIC_PARAM_SESSION_PEER_BIDI_STREAM_COUNT, sizeof(PeerStreamCount), &PeerStreamCount));
             EXIT_ON_FAILURE(MsQuic->SetParam(Session, QUIC_PARAM_LEVEL_SESSION, QUIC_PARAM_SESSION_PEER_UNIDI_STREAM_COUNT, sizeof(PeerStreamCount), &PeerStreamCount));
-        }
 
-        free(AlpnBuffer.Buffer);
+        } else {
+            QUIC_BUFFER AlpnBuffer;
+            AlpnBuffer.Length = (uint32_t)strlen(Settings.AlpnPrefix) + 1; // You can't have more than 2^8 SessionCount. :)
+            AlpnBuffer.Buffer = (uint8_t*)malloc(AlpnBuffer.Length);
+            EXIT_ON_NOT(AlpnBuffer.Buffer);
+            memcpy(AlpnBuffer.Buffer, Settings.AlpnPrefix, AlpnBuffer.Length);
+
+            for (uint32_t i = 0; i < SessionCount; i++) {
+
+                AlpnBuffer.Buffer[AlpnBuffer.Length-1] = (uint8_t)i;
```

Comment:
Shouldn’t this be (uint8_t)(i + 0x30) so at least the ALPN is always ascii?

Reply:
There is no requirement that the ALPN be ASCII.

Reply:
Is it utf8, or encoding agnostic? If spec requires utf8, then this will generate invalid code points? Also, if encoding agnostic what about embedded nulls, which this could put in.

Reply:
It's a raw byte buffer. No string encoding. The fact that ASCII characters are generally used (usually without a NULL) is beside the point.